app/reports/mi-report/get-currency.js

```diff
@@ -0,0 +1,14 @@
+const { PAYMENT_ENRICHED } = require('../../constants/events')
+const { UNKNOWN } = require('../../constants/unknown')
+
+const getCurrency = (events) => {
+  const enrichmentEvent = events.find(event => event.type === PAYMENT_ENRICHED)
+  if (enrichmentEvent) {
+    return enrichmentEvent.data.currency
+  }
+  return events[0]?.data.currency ?? UNKNOWN
```

Comment:
What do the '?' do here exactly?

Reply:
It allows optional chaining -> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

So if we've got no enrichment event, it will try to use the first event.  But if there is no first event then `events[0].data` would throw an error as `events[0]` is undefined.  The `?.` would just mean it doesn't bother trying to evaluate `data` and instead just return undefined.  Which at the end of the line we are coalescing to `UNKNOWN`.  Just as if `currency` wasn't populated.

Reply:
Will definitely have a further read into that optional chaining